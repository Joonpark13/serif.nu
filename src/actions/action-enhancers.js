import { currentTermSelector } from 'selectors';

export const ENHANCE_WITH_CURRENT_TERM_ID = Symbol('ENHANCE WITH CURRENT TERM');
const termIdEnhancer = {
  id: ENHANCE_WITH_CURRENT_TERM_ID,
  mapState: state => ({ currentTermId: currentTermSelector(state).get('id') }),
};

const enhancers = [termIdEnhancer];
export default enhancers;
