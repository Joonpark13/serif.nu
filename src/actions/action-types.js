export const GET_SEARCH_RESULTS_REQUEST = 'GET_SEARCH_RESULTS_REQUEST';
export const GET_SEARCH_RESULTS_SUCCESS = 'SEARCH_RESULTS_SUCCESS';
export const GET_SEARCH_RESULTS_FAILURE = 'SEARCH_RESULTS_FAILURE';
export const CLEAR_SEARCH_RESULTS = 'CLEAR_SEARCH_RESULTS';
export const GET_SECTIONS_REQUEST = 'GET_SECTIONS_REQUEST';
export const GET_SECTIONS_SUCCESS = 'GET_SECTIONS_SUCCESS';
export const GET_SECTIONS_FAILURE = 'GET_SECTIONS_FAILURE';
export const SET_CURRENT_COURSE_NAME = 'SET_CURRENT_COURSE_NAME';
export const VIEW_SEARCH = 'VIEW_SEARCH';
export const ADD_SECTION = 'ADD_SECTION';
export const REMOVE_SECTION = 'REMOVE_SECTION';
export const GET_SCHOOLS_REQUEST = 'GET_SCHOOLS_REQUEST';
export const GET_SCHOOLS_SUCCESS = 'GET_SCHOOLS_SUCCESS';
export const GET_SCHOOLS_FAILURE = 'GET_SCHOOLS_FAILURE';
export const UPDATE_SEARCH_INPUT = 'UPDATE_SEARCH_INPUT';
export const FETCH_SEARCH_INDEX = 'FETCH_SEARCH_INDEX';
export const FETCH_SEARCH_INDEX_SUCCESS = 'FETCH_SEARCH_INDEX_SUCCESS';
export const FETCH_SEARCH_INDEX_FAILURE = 'FETCH_SEARCH_INDEX_FAILURE';
export const FETCH_SUBJECTS_REQUEST = 'FETCH_SUBJECTS_REQUEST';
export const FETCH_SUBJECTS_SUCCESS = 'FETCH_SUBJECTS_SUCCESS';
export const FETCH_SUBJECTS_FAILURE = 'FETCH_SUBJECTS_FAILURE';
