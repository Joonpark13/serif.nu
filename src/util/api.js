/* eslint-disable import/prefer-default-export */
export function searchURL(query) {
  return `${process.env.API_URL}/search?term_id=4720&search_query=${query}`;
}

export function sectionsURL(termId, schoolAbbv, subjectAbbv, courseAbbv) {
  return `${process.env.API_URL}/sections?term_id=${termId}&school_abbv=${schoolAbbv}&subject_abbv=${subjectAbbv}&course_abbv=${courseAbbv}`;
}
