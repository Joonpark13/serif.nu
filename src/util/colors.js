export const northwesternPurple = '#4E2A84';
export const northwesternBrightOrange = '#FFC520';
export const northwesternRichBlack10 = '#D8D6D6';
