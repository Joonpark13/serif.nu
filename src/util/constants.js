/* eslint-disable import/prefer-default-export */
export const CURRENT_TERM = '4730';
