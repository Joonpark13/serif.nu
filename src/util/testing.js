/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
