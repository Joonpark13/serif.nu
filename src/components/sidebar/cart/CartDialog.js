import React, { useState } from 'react';
import { Dialog, DialogActions, DialogTitle, Button } from '@material-ui/core';
import { useDispatch } from 'react-redux';
import { removeAllClasses } from 'actions';

export default function CartDialog() {
  const [open, setOpen] = useState(true);
  const dispatch = useDispatch();

  const handleRemoveAll = () => {
    dispatch(removeAllClasses());
    setOpen(false);
  };
  const handleClose = () => {
    setOpen(false);
  };

  return (
    <Dialog
      open={open}
      onClose={handleClose}
    >
      <DialogTitle>Are you sure you want to remove all classes?</DialogTitle>
      <DialogActions>
        <Button onClick={handleRemoveAll}>
            Remove
        </Button>
        <Button onClick={handleClose}>
            Cancel
        </Button>
      </DialogActions>
    </Dialog>
  );
}
