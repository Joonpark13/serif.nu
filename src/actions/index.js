import * as actionTypes from './action-types';

export function fetchSearchIndex(termId) {
  return {
    type: actionTypes.FETCH_SEARCH_INDEX,
    termId,
  };
}

export function fetchSearchIndexSuccess(searchIndex) {
  return {
    type: actionTypes.FETCH_SEARCH_INDEX_SUCCESS,
    searchIndex,
  };
}

export function fetchSearchIndexFailure() {
  return {
    type: actionTypes.FETCH_SEARCH_INDEX_FAILURE,
  };
}

export const clearSearchResults = () => ({
  type: actionTypes.CLEAR_SEARCH_RESULTS,
});

export function getSectionsRequest(termId, schoolId, subjectId, courseId) {
  return {
    type: actionTypes.GET_SECTIONS_REQUEST,
    termId,
    schoolId,
    subjectId,
    courseId,
  };
}

export const getSectionsSuccess = sections => ({
  type: actionTypes.GET_SECTIONS_SUCCESS,
  sections,
});

export const getSectionsFailure = () => ({
  type: actionTypes.GET_SECTIONS_FAILURE,
});

export const setCurrentCourseName = courseName => ({
  type: actionTypes.SET_CURRENT_COURSE_NAME,
  courseName,
});

export const viewSearch = () => ({
  type: actionTypes.VIEW_SEARCH,
});

export const addSection = section => ({
  type: actionTypes.ADD_SECTION,
  section,
});

export const removeSection = sectionId => ({
  type: actionTypes.REMOVE_SECTION,
  sectionId,
});

export const getSchoolsRequest = () => ({
  type: actionTypes.GET_SCHOOLS_REQUEST,
});

export const getSchoolsSuccess = schools => ({
  type: actionTypes.GET_SCHOOLS_SUCCESS,
  schools,
});

export const getSchoolsFailure = () => ({
  type: actionTypes.GET_SCHOOLS_FAILURE,
});

export const getSearchResultsRequest = searchInput => ({
  type: actionTypes.GET_SEARCH_RESULTS_REQUEST,
  searchInput,
});

export const getSearchResultsSuccess = searchResults => ({
  type: actionTypes.GET_SEARCH_RESULTS_SUCCESS,
  searchResults,
});

export const getSearchResultsFailure = () => ({
  type: actionTypes.GET_SEARCH_RESULTS_FAILURE,
});

export const updateSearchInput = searchInput => ({
  type: actionTypes.UPDATE_SEARCH_INPUT,
  searchInput,
});

export const fetchSubjectsRequest = schoolId => ({
  type: actionTypes.FETCH_SUBJECTS_REQUEST,
  schoolId,
});

export const fetchSubjectsSuccess = subjects => ({
  type: actionTypes.FETCH_SUBJECTS_SUCCESS,
  subjects,
});

export const fetchSubjectsFailure = () => ({
  type: actionTypes.FETCH_SUBJECTS_FAILURE,
});

export const changeBrowseLevel = browseLevel => ({
  type: actionTypes.CHANGE_BROWSE_LEVEL,
  browseLevel,
});
