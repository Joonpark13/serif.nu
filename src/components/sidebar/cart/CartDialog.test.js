import React from 'react';
import { shallow } from 'enzyme';
import { mockUseSelector, mockUseDispatch } from 'util/testing';
import { Button } from '@material-ui/core';
import { removeAllClasses } from 'actions';
import CartDialog from './CartDialog';


describe('CartDialog', () => {
  beforeEach(() => {
    mockUseSelector([]);
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <CartDialog />,
    );

    expect(wrapper.get(0)).toMatchSnapshot();
  });

  it('removes all classes when clicked on "remove"', () => {
    mockUseSelector([{ id: '123' }, { id: '123' }]);

    const dispatchMock = mockUseDispatch();
    const wrapper = shallow(<CartDialog />);
    wrapper.find(Button).first().simulate('click');

    expect(dispatchMock).toHaveBeenCalledWith(removeAllClasses());
  });

  it('closes the modal when clicked on "cancel"', () => {
    mockUseSelector([{ id: '123' }, { id: '123' }]);

    const wrapper = shallow(<CartDialog />);
    wrapper.find(Button).at(1).simulate('click');
    expect(wrapper.find(CartDialog)).toHaveLength(0);
  });
});
