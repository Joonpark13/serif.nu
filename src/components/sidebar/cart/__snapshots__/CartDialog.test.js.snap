// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartDialog renders correctly 1`] = `
<ForwardRef(WithStyles)
  onClose={[Function]}
  open={true}
>
  <ForwardRef(WithStyles)>
    Are you sure you want to remove all classes?
  </ForwardRef(WithStyles)>
  <ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      onClick={[Function]}
    >
      Remove
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      onClick={[Function]}
    >
      Cancel
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
</ForwardRef(WithStyles)>
`;
